include/dxc/Support/DxcOptToggles.h

```diff
@@ -0,0 +1,72 @@
+///////////////////////////////////////////////////////////////////////////////
+//                                                                           //
+// DxcOptToggles.h                                                           //
+// Copyright (C) Microsoft Corporation. All rights reserved.                 //
+// This file is distributed under the University of Illinois Open Source     //
+// License. See LICENSE.TXT for details.                                     //
+//                                                                           //
+// Helper code for representing -opt-disable, -opt-enable, -opt-select       //
+// options                                                                   //
+//                                                                           //
+///////////////////////////////////////////////////////////////////////////////
+
+#pragma once
+
+#ifndef LLVM_HLSL_DXC_OPT_TOGGLES_H
+#define LLVM_HLSL_DXC_OPT_TOGGLES_H
+
+#include "llvm/ADT/StringRef.h"
+#include <map>
+#include <set>
+#include <string>
+
+namespace hlsl {
+
+namespace options {
+
+struct Toggle {
+  llvm::StringRef Name;
+  bool Default = false;
+  Toggle(llvm::StringRef Name, bool Default) : Name(Name), Default(Default) {}
```

Comment:
Using `StringRef` means that the memory is not copied and must be valid for the lifetime of the Toggle value. Since it is always passed a literal sting, this should be ok, but would potentially be a problem if it's using a dynamic string name.